apps/src/blockly/addons/cdoUtils.ts

```diff
@@ -443,6 +445,37 @@ export function getCode(workspace: WorkspaceSvg, getSourceAsJson: boolean) {
   }
 }
 
+/**
+ * Ensure that only a number may be entered.
+ * @param {string} text The user's text.
+ * @returns {?string} A string representing a valid number, or null if invalid.
+ *   Returns 0 for null or empty string.
+ * @static
+ */
+export function numberValidator(text: string): string | null {
+  text = text || '';
+  // TODO: Handle cases like 'ten', '1.203,14', etc.
+  // 'O' is sometimes mistaken for '0' by inexperienced users.
+  text = text.replace(/O/gi, '0');
+  // Strip out thousands separators.
+  text = text.replace(/,/g, '');
+  const n = parseFloat(text || '0');
+  return isNaN(n) ? null : String(n);
+}
+
+/**
+ * Ensure that only a nonnegative integer may be entered.
+ * @param {string} text The user's text.
+ * @returns {?string} A string representing a valid int, or null if invalid.
+ * @static
+ */
+export function nonnegativeIntegerValidator(text: string): string | null {
+  const n = numberValidator(text);
+  if (n) {
+    return String(Math.max(0, Math.floor(parseFloat(n))));
```

Comment:
Am I correct in reading that this turns negative numbers into 0? If so can you update the description? I read the description as if the number is negative we return `null`.

Reply:
Add a clarifying example to the description! It will return `null` if the input isn't a valid number, ex. `'abc'`.